src/dataset-builder/ConceptSearch.ts

```diff
@@ -34,15 +35,23 @@ const DebouncedTextInput = withDebouncedChange(TextInput);
 export const ConceptSearch = (props: ConceptSearchProps) => {
   const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
     props;
-  const [search, setSearch] = useState<string>(initialSearch);
+  const [searchText, setSearchText] = useState<string>(initialSearch);
   const [cart, setCart] = useState<Concept[]>(initialCart);
   const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
 
-  useEffect(() => {
+  // when mounting ConceptSearch, we want to retrieve a list of searchConcepts
+  useOnMount(() => {
     void searchConcepts(() => {
-      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root);
     });
-  }, [search, datasetId, domainOption.root, searchConcepts]);
+  });
+  useEffect(() => {
+    if (searchText.length > 2) {
```

Comment:
I think this is not quite correct, as you want to refresh the view of all items if the search string is empty. Something like
```suggestion
    if (searchText.length === 0 || searchText.length > 2) {
```

Reply:
Yep!